conda-store-server/conda_store_server/_internal/server/views/conda_store_ui.py

```diff
@@ -3,18 +3,34 @@
 # license that can be found in the LICENSE file.
 
 from fastapi import APIRouter, Depends, Request
+from fastapi.responses import HTMLResponse
 
 from conda_store_server._internal.server import dependencies
 
-router_conda_store_ui = APIRouter(tags=["conda-store-ui"])
+router_conda_store_ui = APIRouter(
+    tags=["conda-store-ui"],
+    # Provide the response class so that it can show up in the autogenerated API
+    # docs. The docs show the endpoints as returning HTML rather than JSON.
+    default_response_class=HTMLResponse,
+)
 
 
+# Yes, "{path:path}"" makes "/" redundant. But if we don't define "/", then we
+# always have to pass `path` to the `url_for` function like so:
+# url_for("get_conda_store_ui", path="")
 @router_conda_store_ui.get("/")
+@router_conda_store_ui.get("{path:path}")
```

Comment:
@peytondmurray I'm not entirely sure why, but this seems to be the line that is causing at least some of the CI checks to fail.

If you remove this line and go to localhost:8080/api/v1/namespace, you'll see that it redirects to same URL but with a forward slash at the end. If you then put this line back in and go to the same URL, you'll notice that it tries to load the client app. You get an HTML page back that tries to load the client app at that URL, but the client renders an error since it doesn't recognize that route (api/v1/namespace).

Any idea how to fix it?

Reply:
Recap

With line 22 removed:

- localhost:8080/api/v1/namespace -> localhost:8080/api/v1/namespace/
- localhost:8080/api/v1/namespace/ -> JSON containing list of namespaces

With line 22 in place:

- localhost:8080/api/v1/namespace -> index.html for client app
- localhost:8080/api/v1/namespace/ -> JSON containing list of namespaces

Reply:
This seems to suggest to me if FastAPI tries all the routes and cannot find a match, then it will add a trailing slash as a last-ditch attempt to not return a 404. 

I see there is an option `redirect_slashes` for the [FastAPI class](https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI) whose default value is True. 

I will have to look at the FastAPI source code to see how and when it uses that option.

I wonder if the solution is to explicitly map both routes to the same handler. The following would code would allow the user to get the internal admin UI via `/admin` or via `/admin/`

```
@router_ui.get("")
@router_ui.get("/")
async def ui_list_environments(
```

Reply:
I don't have context on this, but it seems like this is intended behavior on the FastAPI side of things. Others have [wondered](https://github.com/fastapi/fastapi/discussions/7298) about how to prevent this and [about the impacts of redirection on http[s]](https://github.com/fastapi/fastapi/discussions/9328).